group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/RangeAssignorTest.java

```diff
@@ -0,0 +1,523 @@
+
+        Map<String, AssignmentMemberSpec> members = new TreeMap<>();
+
+        // Let initial subscriptions be A -> T1, T2 // B -> T2 // C -> T2, T3
+        // Change the subscriptions to A -> T1 // B -> T1, T2, T3 // C -> T2
+
+        Map<Uuid, Set<Integer>> currentAssignmentForA = new HashMap<>();
+        currentAssignmentForA.put(topic1Uuid, new HashSet<>(Arrays.asList(0, 1, 2)));
+        currentAssignmentForA.put(topic2Uuid, Collections.singleton(0));
+        members.put(consumerA, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Collections.singletonList(topic1Uuid),
+            currentAssignmentForA
+        ));
+
+        Map<Uuid, Set<Integer>> currentAssignmentForB = new HashMap<>();
+        currentAssignmentForB.put(topic2Uuid, Collections.singleton(1));
+        members.put(consumerB, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid, topic3Uuid),
+            currentAssignmentForB
+        ));
+
+        Map<Uuid, Set<Integer>> currentAssignmentForC = new HashMap<>();
+        currentAssignmentForC.put(topic2Uuid, Collections.singleton(2));
+        currentAssignmentForC.put(topic3Uuid, new HashSet<>(Arrays.asList(0, 1)));
+        members.put(consumerC, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Collections.singletonList(topic2Uuid),
+            currentAssignmentForC
+        ));
+
+        AssignmentSpec assignmentSpec = new AssignmentSpec(members, topics);
+        GroupAssignment computedAssignment = assignor.assign(assignmentSpec);
+
+        Map<String, Map<Uuid, Set<Integer>>> expectedAssignment = new HashMap<>();
+        // Topic 1 Partitions Assignment
+        expectedAssignment.put(consumerA, mkAssignment(mkTopicAssignment(topic1Uuid, 0, 1)));
+        expectedAssignment.put(consumerB, mkAssignment(mkTopicAssignment(topic1Uuid, 2)));
+        // Topic 2 Partitions Assignment
+        expectedAssignment.get(consumerB).putAll(mkAssignment(mkTopicAssignment(topic2Uuid, 0, 1)));
+        expectedAssignment.put(consumerC, mkAssignment(mkTopicAssignment(topic2Uuid, 2)));
+        // Topic 3 Partitions Assignment
+        expectedAssignment.get(consumerB).putAll(mkAssignment(mkTopicAssignment(topic3Uuid, 0, 1)));
+
+        assertAssignment(expectedAssignment, computedAssignment);
+    }
+
+    private void assertAssignment(Map<String, Map<Uuid, Set<Integer>>> expectedAssignment, GroupAssignment computedGroupAssignment) {
+        for (String memberId : computedGroupAssignment.members().keySet()) {
+            Map<Uuid, Set<Integer>> computedAssignmentForMember = computedGroupAssignment.members().get(memberId).targetPartitions();
+            assertEquals(expectedAssignment.get(memberId), computedAssignmentForMember);
+        }
+    }
+
+    private void assertCoPartitionJoinProperty(GroupAssignment groupAssignment) {
```

Comment:
Is this still useful now that we have `assertAssignment`?

Reply:
we don't need it but we wanted separate property tests right?

Reply:
Yeah, it was because we were not able to use equals. Now that we can use it, I am not sure that this one bring any value. Does it?

Reply:
nop we can remove it, ig whoever wants it later can write it again